src/wallet/init.cpp

```diff
@@ -116,6 +116,9 @@ void WalletInit::Construct(NodeContext& node) const
         LogPrintf("Wallet disabled!\n");
         return;
     }
-    gArgs.SoftSetArg("-wallet", "");
+    WalletLocation default_wallet("wallet.dat");
+    if (default_wallet.Exists()) {
+        gArgs.SoftSetArg("-wallet", "");
```

Comment:
In commit "Do not create default wallet" (6feed9508a225276cc2264b7d19b151a62deecea)

Was experimenting with this PR and would suggest the following fixup: 973bb7e652c4ce10ee536e9967a88c80f6dbe2e0 ([branch](https://github.com/ryanofsky/bitcoin/commits/review.15454.2-edit)). SoftSet doesn't really work here for a persistent setting. I'll backport this fixup to #15937 so the diff here will be smaller, and the only code change this should have to make is adding ` && WalletLocation("").Exists()`

Reply:
re: https://github.com/bitcoin/bitcoin/pull/15454#discussion_r445559622

> Was experimenting with this PR and would suggest the following fixup: [973bb7e](https://github.com/bitcoin/bitcoin/commit/973bb7e652c4ce10ee536e9967a88c80f6dbe2e0) ([tag](https://github.com/ryanofsky/bitcoin/commits/review.15454.2-edit.1))

Feel free to steal updated version of this PR with fix above squashed and newer changes from 15937:
https://github.com/ryanofsky/bitcoin/commits/review.15454.2-edit